src/main/java/com/puppycrawl/tools/checkstyle/checks/design/FinalClassCheck.java

```diff
@@ -170,12 +181,9 @@ public void visitToken(DetailAST ast) {
 
             case TokenTypes.LITERAL_NEW:
                 if (ast.getFirstChild() != null
-                        && ast.getLastChild().getType() == TokenTypes.OBJBLOCK) {
-                    for (ClassDesc classDesc : classes) {
-                        if (doesNameOfClassMatchAnonymousInnerClassName(ast, classDesc)) {
-                            classDesc.registerAnonymousInnerClass();
-                        }
-                    }
+                    && ast.getLastChild().getType() == TokenTypes.OBJBLOCK) {
```

Comment:
indentation

Reply:
Done

Reply:
original indentation was like this, and there is no reason to change it.
Plus, usually second condition is indented, even if it is on same level as first. For your example I would say that indentation should look like
```
a 
/*1 indent */&& b
/*1 indent */&& (c 
/*2 indent */    || d)
/*1 indent */ && e
```